src/main/java/org/prebid/server/bidder/taboola/TaboolaBidder.java

```diff
@@ -0,0 +1,180 @@
+                .publisher(Publisher.builder().id(impExt.getPublisherId()).build())
+                .build();
+
+        return request.toBuilder()
+                .badv(CollectionUtils.isNotEmpty(blockedAdomain) ? blockedAdomain : request.getBadv())
+                .bcat(CollectionUtils.isNotEmpty(blockedAdvCat) ? blockedAdvCat : request.getBcat())
+                .imp(Collections.singletonList(updatedImp))
+                .ext(modifiedExtRequest)
+                .site(newSite)
+                .build();
+    }
+
+    private String buildEndpointUrl(ExtImpTaboola extImpTaboola, String type, String domain) {
+        return endpointTemplate.replace("{{Host}}", domain)
+                .replace("{{MediaType}}", type)
+                .replace("{{PublisherID}}", extImpTaboola.getPublisherId());
+    }
+
+    private static Imp modifyImp(Imp imp, ExtImpTaboola impExt) {
+        Banner banner = imp.getBanner();
+        if (banner != null && impExt.getPosition() != null) {
+            banner = banner.toBuilder().pos(impExt.getPosition()).build();
+        }
+        return imp.toBuilder()
+                .tagid(impExt.getPublisherId())
+                .bidfloor(impExt.getBidFloor())
+                .banner(banner)
+                .build();
+    }
+
+    private ExtRequest modifyExtRequest(ExtRequest extRequest, String pageType) {
+        final ObjectNode adfNode = mapper.mapper().createObjectNode().put("pageType", pageType);
+        return mapper.fillExtension(extRequest, adfNode);
+    }
+
+    private static BidType getBidType(Imp imp) {
+        return imp.getBanner() != null ? BidType.banner : BidType.xNative;
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.withValues(extractBids(httpCall.getRequest().getPayload(), bidResponse));
+        } catch (DecodeException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse);
+    }
+
+    private List<BidderBid> bidsFromResponse(BidRequest bidRequest,
+                                             BidResponse bidResponse) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .flatMap(seatBid -> seatBid.getBid().stream()
+                        .flatMap(bid -> bidRequest.getImp().stream()
+                                .map(imp -> BidderBid.of(bid, getBidType(imp), bidResponse.getCur()))))
```

Comment:
Please, return previous solution.

Also, you can't use same `getBidType` method for response.
Pay attention that when you will throw exception about something wrong in the current bid, you need to catch it, add an error to errors list and continue process other bids.